pkg/platform/platform.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package platform
+
+import "istio.io/istio/pkg/env"
+
+type PlatformType string
+
+const (
+	K8s       PlatformType = "k8s"
```

Comment:
Done. I used a constant named `Empty` for this purpose. Let me know what you think.

Also, is `gcp` good? or is `gke` better?

Reply:
Its a bit confusing since gcp can be GKE or "some k8s distro on GCP compute". And GKE can be "GKE distro on <not GCP cloud/on prem>"...

So overall, I don't know. We could just remove it for now since we don't use it for anything? If not I would go with `gcp` somewhat arbitrarily...

Reply:
In this case I think it's better to leave it there, although in a near term only openshift will be used, it could be useful later.